### PR TITLE
Containerise Bouncer.

### PR DESCRIPTION
**Pull request by sengi**
Should be no functional change for the existing deployment on EC2.

- Add Dockerfile.
- Add some missing `require`s to `boot.rb` that we were getting away without when running under Unicorn.
- Use bootsnap to reduce the resource usage spike on startup.
- Add GitHub Actions workflows for CI (lint and `rake spec`) and build/push to Kubernetes.

Tested: builds and runs on Docker Desktop.

### CODE REVIEW

**Review by theseanything — APPROVED** [review]

**Comment by sengi** [comment]
I'll wait until the morning before merging this one since it's push-on-green.

**Comment by sengi** [comment]
Realised I needed the workflows to build and push to k8s, so there's a bit more now in case you wanna take another look.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - main
+    paths-ignore:
+      - "Jenkinsfile"
+      - ".git**"
+  pull_request:
+
+jobs:
+  lint-ruby:
+    name: Lint Ruby
+    uses: alphagov/govuk-infrastructure/.github/workflows/rubocop.yml@main
```
Comment (theseanything) [inline .github/workflows/ci.yml:16]: Nice! By the looks of it we've not been running Rubocop in Jenkins (as it's not part of the [default rake task](https://github.com/alphagov/bouncer/blob/69a1cf1d1e6ce52dca487c3a91cdd8211db284a5/Rakefile#L7)).

(Also why there was a bunch linting errors to fix)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Initialize database
+        env:
+          RACK_ENV: test
+          TEST_DATABASE_URL: ${{ steps.setup-postgres.outputs.db-url }}
+        run: bundle exec rake db:reset
```
Comment (theseanything) [inline .github/workflows/ci.yml:38]: Once we're no longer using Jenkins - we won't need to "reset" the db anymore and can remove the `dropdb` command.

**Review by theseanything — APPROVED** [review]

**Comment by sengi** [comment]
Awesome, thanks for bearing with me there!

**Pull request merged**